client/src/components/UploadComponent.js

```diff
@@ -0,0 +1,339 @@
+import React, { useState, useRef } from "react";
+import styled from "styled-components";
+
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faUpload, faTrash } from "@fortawesome/free-solid-svg-icons";
+
+const KILO_BYTES_PER_BYTE = 1000;
+const DEFAULT_MAX_FILE_SIZE_IN_BYTES = 500000;
+
+const convertNestedObjectToArray = (nestedObj) =>
+  Object.keys(nestedObj).map((key) => nestedObj[key]);
+
+const convertBytesToKB = (bytes) => Math.round(bytes / KILO_BYTES_PER_BYTE);
+
+const FileUpload = ({
+  label,
+  updateFilesCb,
+  maxFileSizeInBytes = DEFAULT_MAX_FILE_SIZE_IN_BYTES,
+  ...otherProps
+}) => {
+  const fileInputField = useRef(null);
+  const [files, setFiles] = useState({});
+
+  const handleUploadBtnClick = () => {
+    fileInputField.current.click();
+  };
+
+  const addNewFiles = (newFiles) => {
+    for (let file of newFiles) {
+      if (file.size < maxFileSizeInBytes) {
+        if (!otherProps.multiple) {
+          return { file };
+        }
+        files[file.name] = file;
+      }
+    }
```

Comment:
Does this actually work or maybe throw an error in the console? Usually the rule of hooks suggests not to use hooks inside a javascript function but only from a React functional component